fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/service/LoanScheduleAssembler.java

```diff
@@ -466,7 +466,11 @@ private LoanApplicationTerms assembleLoanApplicationTermsFrom(final JsonElement
         final boolean isPrincipalCompoundingDisabledForOverdueLoans = this.configurationDomainService
                 .isPrincipalCompoundingDisabledForOverdueLoans();
 
-        final boolean isDownPaymentEnabled = loanProduct.getLoanProductRelatedDetail().isEnableDownPayment();
+        boolean isDownPaymentEnabled = loanProduct.getLoanProductRelatedDetail().isEnableDownPayment();
```

Comment:
Please add validation error when user tries to enable down payment but it was not configured on loan product!
Ignoring the field that was sent is wrong!

Reply:
Done! new validation method implemented in the LoanScheduleValidator